server/src/main/java/server/BazelServices.java

```diff
@@ -187,4 +189,19 @@ public void connect(LanguageClient client) {
     public CompletableFuture<CompletionItem> resolveCompletionItem(CompletionItem unresolved) {
         return CompletableFuture.completedFuture(unresolved);
     }
+
+    @Override
+    public CompletableFuture<List<? extends TextEdit>> formatting(DocumentFormattingParams params) {
+        logger.info("Formatting request received");
+        Buildifier buildifier = new Buildifier();
+        // Formatting is done through the buildifier. We must verify that the client has buildifier installed.
+        if (buildifier.exists()) {
+            FormattingProvider formattingProvider = new FormattingProvider(DocumentTracker.getInstance(), buildifier);
+            return formattingProvider.getDocumentFormatting(params);
+        } else {
+            // Display a popup indicating the client does not have buildifier installed.
+            languageClient.showMessage(new MessageParams(MessageType.Info, "Buildifier executable not found.\nPlease install buildifier to enable file formatting."));
+            return CompletableFuture.completedFuture(new ArrayList<TextEdit>());
```

Comment:
I think you need a import java.util.ArrayList; at the top of this file to use an ArrayList

Reply:
Oh right. Thanks for double checking that, I thought I had called build after making the final changes but I guess I was wrong.

Reply:
Oh yeah that was because I merged develop into our branch after we'd made our changes. So that explains why I missed this. Thanks for catching it!